test/e2e/util_test.go

```diff
@@ -595,6 +597,25 @@ func updateInfrastructureConfigSpecWithRetryOnConflict(t *testing.T, name types.
 	})
 }
 
+// updateInfrastructureStatus updates the Infrastructure status by applying
+// the given update function to the current Infrastructure object.
+func updateInfrastructureConfigStatusWithRetryOnConflict(configClient *configclientset.Clientset, updateFunc func(*configv1.Infrastructure) *configv1.Infrastructure) error {
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
The `retry.DefaultRetry` is only 10ms https://pkg.go.dev/k8s.io/client-go/util/retry#pkg-variables.  Why not use `wait.PollUntilContextTimeout(context.Background()` and allow it to loop on conflict for a configured amount of time?

Reply:
Sure, we can use `wait.PollUntilContextTimeout(context.Background()` here as well. I've updated the logic in the latest changes. Thanks!